### PR TITLE
Add Conventional Commit linter to CI

### PR DESCRIPTION
**Pull request by KingMob**

### CODE REVIEW

**Comment by KingMob** [comment]
@ships Can you check my GH Action, and let me know if I got the nix setup part right? I'm unclear on when I can use `nix develop --command my-command` and when I use `nix develop --command bash -c "my-command"`

I got a lot of 429 Too Many Requests errors until the cache filled up. Is there a way around that?

Finally, I ran `nix flake update`, but if I wanted to _only_ add node v20, how would I add that to the flake.lock while leaving the other packages unchanged? I'd like to operate under the if-it-aint-broke-dont-fix-it principle when possible.

@zane I set the rules to match your existing style. Generally speaking, do you want this GH Action to warn people, or actually block merging? And are their other repos this should be applied to?

**Comment by KingMob** [comment]
> Basically the ones that are already using this commit style.

@zane OK, looks to me like `Gen.clj`, `inference`, `publish`, `structure-learning`, and `GenSQL.Gen.clj.adapter`. Umm,`cloud-builders`, yay or nay?

**Comment by KingMob** [comment]
@ships Does the nix stuff look right? Do I need the `bash -c` or no?

File: .github/workflows/commit-linter.yaml
Comment (ships) [inline .github/workflows/commit-linter.yaml:38]: @KingMob This all looks correct. One modification might be to add `-V` flag to `commitlint` so we  can introspect what was checked  as recommended in your update to the README
```
$  npx commitlint -f head~ -t head -V
⧗   input: feat: Add Conventional Commits linter for CI
✔   found 0 problems, 0 warnings
```

File: flake.nix
Comment (ships) [inline flake.nix:57]: Notably, there are also `nodePackages` available in nixpkgs that we currently don't use (preferring to make you `npm install`). But in theory we could add those packages here too. This change makes sense to me as is though.

**Review by ships — APPROVED** [review]
LGTM, commented on optional verbosity flag for human readers.

File: .github/workflows/commit-linter.yaml
Comment (KingMob) [reply .github/workflows/commit-linter.yaml:38]: OK, will add

File: flake.nix
Comment (KingMob) [reply flake.nix:57]: Ahh, I wasn't sure. I see it in nixpkgs now.

I feel like incompleteness is always such a problem. Like, if there's any package not available in nix, you have to fall back to package.json anyway, and then... what? Merges and conflicts? Human sacrifice, dogs and cats living together, mass hysteria!

**Comment by KingMob** [comment]
@zane After merging, I'd like to give it a bit of time, make sure it fits smoothly into our workflow, and then roll out to other repos after a week or two.

**Pull request merged**
